### DISCUSSION TITLE
Proof of Knowledge as a user acquisition channel for Akash vLLM

### DISCUSSION THREAD

**Discussion by JBate77** [akash-network/community#656] [AI - Artificial Intelligence]
# Introduction
# tl;dr

PoK as a user acquisition channel for Akash usage.

Proof of Knowledge (PoK) is a protocol that faciliates collaborative, incentivised onchain knowledge creation for general deAI use. We maintain a global vector database ("Brains") of embeddings, pulled from individual "Brains", each with their own context and intelligence. The individual Brains are maintained by onchain associations, DAOs, businesses, conferences, creators... anyone that wants to store data in a way that is queryable via natural language. PoK provides the infra for this data collection and curation, a tracking and an incentive layer for the use of those data, and a global distribution mechanism via advanced RAG LLM search - all as a public good.

# Benefit for Akash

We are using vLLM instances for each "Brain" and want to become an effective "middleman" for Akash, by recommending our users to spin up a vLLM model, as the primary distributed model hosting service. We are asking for a grant to help kickstart our efforts and to deepen collaborations. With the planned adoption scale of PoK, Akash could see significant usage from the PoK community and its users, each hosting an instance for their Brain when required and spinning down when they are not using it. PoK maintains an embedding profile of each Brain permanently, even when their instance is spun down.

# Context
### PoK in the deAI Stack
The decentralised AI stack consists of many pillars: compute, models, training, storage, agents, incentives and -- *data* --.
Data collection and curation difficult and messy tasks and, as such, few projects are tackling this pivotal aspect of the ecosystem. Many data exist on the internet and are scraped for AI training and retrieval but with the core issue of provenance, value distribution and importantly, ownership. These are issues that deAI promises to solve.

Optimal data for the deAI future has the following qualities:
- structured and optimised for machine readability (**_structured_**);
- attested to the original author with an immutable signature (**_attested_**);
- possible to implement tracking whenever the data are used or queried by AI (**_tracked_**);
- interoperable with other data types and AI models (**_interoperable_**).

If data can possess these four qualities, it can form a robust foundation for the deAI ecosystem, as a primary unit of knowledge for the distribution of information, coordination of people, groups and agents and accumulation of a novel type of reputation, based on the usefulness of data for RAG LLM.

### PoK Rollout
Currently, PoK has been deployed in offchain, private environments for coordinating groups both online and irl at events - each a "Brain". The system is effectively embedding chatstreams, documents and specific datasets. This supports internal coordination of people and tasks and the embeddings can be queried using natural language to great effect. The next stage is to connect the individual Brains together such that users can receive information about the other Brains, to further improve knowledge sharing and coordination. Then, we connect the Brains to our onchain components that attest data units to wallet addresses and distribute points based on their usage. 

### Team (DeSciWorld)
PoK is a public good infrastructure deployed by DeSciWorld. DeSciWorld was first created in 2021 and has been building agnostic public goods tools for the DeSci space. We have a dashboard deployed at https://desci.world/, an extensive [repository](https://github.com/DeSciWorldDAO/awesome-desci) of DeSci resources and have hosted a dozen conferences around the world with the aim of educating and onboarding DeScientists.

The team consists of scientists, engineers, entrepreneurs and researchers. Team members are listed below:
Joshua Bate - Founder - Twitter: @jb87ua
At0x.eth - CTO - Twitter: @at0x_eth
Dr. Jelani Clarke, PhD - Chief Scientist Relations Officer - https://www.linkedin.com/in/jelani-clarke-ph-d-920b1080/
Carolina Menchaca, MsC - Chief Researcher - https://www.linkedin.com/in/carolina-menchaca-69723b1a/
Luis Maumejean - Head of LATAM - https://www.linkedin.com/in/lemg/
Carlos DiMatteo - Senior Engineer 
Raquel Raigal - Junior Engineer

# Timeline
The timeline for this project is as follows:
Open Discussions: Starting August 2024
Governance Proposal: Through September 2024
Design Phase: Ongoing as of today
First 10 Brains deployed: End of October 2024
Full mainnet deployment with permissionless interface and onchain points: December 2024
Programme end: Q1 2025
_Updated referral and partnership arrangement hereafter_
Note: This is subject to change based on feedback

# Budget
The proposed budget is consists of [Akash costs] (url) and Team Costs: ($50,000 + $25,000)

Usage costs are to help us subsidise the initial rollout of the Brains, giving free AKT to certain users as a donation to their organisation, to host the vLLM without cost. In the future, we will stop the subsidy and future users must decide to purchase AKT to fund their own models.

Team Costs are to assist the team during this period as we are yet to raise large capital funding and want to continue building with the full team's effort. 

### Disbursement
The Team Costs will be disbursed immediately to the DeSciWorld Multisig at: 
The Akash Costs funds will be divided into 3 tranches:
1. Internal use - $15,000 for the internal LLM provisioning so that we can have multiple different models for different tasks - this would be the "elder brain"

2. Grants - $25,000 to be distributed within 6 months of receipt to vetted and approved projects to spin up community nodes with their own LMM instance (because it can be fine tuned)

3. Reserve - $10,000 will be reserved in case of unexpected compute demand 

### Continuted Collaboration
If the proposal is successful and PoK becomes a high quality user acquisition partner for Akash, we hope to revisit a discussion into 2025 that allows for continuous subsidy of aspects of PoK, or referral revenue share to DeSciWorld, depending on results and returns for Akash.

**Comment by dominikusbrian** [comment]
This is interesting concept, if I may ask and learn a bit more, how this solution compared to the OriginTrail Decentralized Knowledge Graph and Knowledge Mining approach ?


Very supportive toward this idea especially on having the Grants aspect of promoting uses of Fine-tuning own LLM instance. 
Recently are working with this open-source AI Scientist by Sakana AI https://github.com/SakanaAI/AI-Scientist that allows, for producing very coherent, exceptional research paper, event with the most standard models. They are now moving toward developing and testing usage of open models which would be amazing, but that requires some H100s resources or something as powerful to support the test. 

This kind of grants will be of utmost help to many researcher and scientist that have great ideas and capability to test them, but doesn't have enough resources to get expensive GPUs on their own. Renting dedicated server or bare metals and setting up everything + maintaining the support infra on one's own is also very challenging especially for non IT/Network expert, which most scientist aren't. 

Looking forward to further updates on this. 

Would certainly love to be looped in for the use-case testing, I can submit several of our own project proposal to demonstrate what kind of fine tuning and use of PoK can be spinned out this Akash x DeSci.World tech Integration.

**Reply by JBate77** [reply]
Thanks for the question Domi 🙂

First off, its good to note that the Decentralised Knowledge Graph (DKG) space is pretty new and wide open for different models to find use cases. We are actually good friends with the OriginTrail team and plan to collaborate once our product is more mature. Our DKG (PoK) can exist as a parachain within the OriginTrail ecosystem, as a source of knowledge/info for users that are querying their DKG.

PoK is a DKG that provides a universal incentive (points) layer that:

a) is capable of more throughput regarding ingesting lots of data, including chatstreams, and outputting many queries. This is really important for the community "Brains" concept, as each community will not want to moderate their input and are likely to have fun with the bot giving it a lot of prompts.

b) as such, PoK is optimised for usefulness, coordinating people and resources and interesting bot interactions

c) would interoperate with OriginTrail and other solutions 

d) would create a primitive on ETH L1 that can be composed on by other protocols, including diverse sets of apps or games

**Reply by dominikusbrian** [reply]
Thanks @JBate77  for the detailed explanations. 

That being said, would really interested in exploring the tech module you and the team is building as they become available. 

Regarding the ： 

> First 10 Brains deployed: End of October 2024

Is the projects that aim to deploy this 10 brains has already been selected ahead of times? or the scheme will be an open Call to Action that allows any possible client of the platform to start deploying as they see fit and provided the compute resources for the vLLM server can accomodate the demand?


In addition of the great potential this PoK technology can bring about, I also foresee the possible challenge toward realizing the four qualities for deAI described-- Structured, Attested, Tracked, Interoperable. In the general data science and scientific data community it has been generally been accepted that a great shared database or reproducible data should have the FAIR properties (Findability, Accessibility, Interoperability, Reproducibility). 

One in particular-- Accessibility-- may somewhat be a challenge toward getting the PoK to be able providing the "best" truth. Since as we all know it the current information sea of the world still is largely divided between institution Paywalled quality information and publicly available open information. Just to give an instance that may led to possible challenge, with the attestation to original author, the system might have a good chance to achieve this attestation of original author only up to "publicly known" original author, whereas the original author itself, if their work has been around before the era of public information (which many still are), will indefinitely have slighlty or maybe significantly reduced chance from getting their original contribution recognized properly. This "mistake" will then naturally propagate throughout the system forward, due to the tracked nature of the deAI. 

I understood that many of the components and how the PoK system would run is still much being designed carefully and under development. That being said, would be nice to have a regular update system in place, for how these "Brains" performed over time. Maybe through a monthly blog post or something else that make sure the knowledge from this development and attempts to push deAI is well documented and will be available for whoever people or organization that going to build on top of it.

**Reply by Ataxia123** [reply]
thanks for the very thorough response!

The main goal of this system is to "tease" out information (and more importantly the meta-information) from gardens by providing an incentive for it to prove its worthyness. I always say what we want to make the least opinionated metric that can be designed as a ground truth (being: how many times has this been retrieved?)


The main goal of this would be to provide a credible starting point for ideas even before they coalesce into something productive they could be traced back to their first instance on the graph. Now to your point we do want to recognize "pre-graph" knowledge and we intend to set aside a percentage of the issuance for retroactive rewarding of data that is verified by the original author. On that note one of the first tasks of the system would be to define its own monetary policy and evolve that over time to cover all these edge cases. we are going to propose a starting setup with a 4 year "presidential" term after which we will reelect a governance team.

All of these ideas are our best case scenario architecture which we've been trialing/evolving over time. Im happy to add more brains to the conversation to get this to be a really robust primitivefor everyone to build on

**Reply by dominikusbrian** [reply]
This sounds heading to the right direction. 

Not quite sure how's the scope of authority this rotating "president"-led governance team would have relate to the current proposal. But yes I got the notion on how reward distribution will be set down the road will be strongly associated with the monetary policy of the ecosystem. 

Just recently show the GraphRAG tech open-sourced by Microsoft, 
link to paper here : https://arxiv.org/abs/2404.16130
link to GitHub here : https://github.com/microsoft/graphrag?tab=readme-ov-file

This GraphRAG and others network relationship structure could perhaps be of use in determining the "center-of-mass" that are closest to the truth. Because philosophically there's no such thing like absolute truth, truth merely are accepted facts by most people that sort of agreed on a certain version of thing. The graph will help quantify the  multidimensional gravity of which node of knowledge or set of knowledge best represent something close to "truth". Yes the pre-graph concept would be interesting, because by tracking the network and connectivity before the information nodes coalesce together, one can have a quantifiable degree of similarity. Just like how we work on understanding evolution in biological species. 

One question regarding the "point based on usage" how granular is this point-system ? 
Surely for some "brains" user doesn't necessarily need access to the whole part of it, maybe just relevant part of each specific "brain". Moreover, this point system will somehow translate to economic value like financial cost too right ? 

Back to the main thread of this discussion, so the action plan for this is to simply get the latest version of the proposal on-chain throughout September ?

**Comment by 31trainman** [comment]
I'm happy to lend a hand in anything I can with this project.

**Reply by JBate77** [reply]
Hey, thanks for this! Would love to have as many supporters/collaborators as possible with this project. We're a startup with limited resources and want to scale the tech as quickly as possible. What are your interests/skillsets?

**Reply by 31trainman** [reply]
I have knowledge of creating deployments on Akash Network. As well as multiple connections to akash core team members and core team members of other depin projects.

**Comment by 31trainman** [comment]
I think this has integration potential with your idea https://www.gravwell.io/product-overview?utm_source=google_ads&utm_medium=brand&kw=gravwell&cpn=15507596878&utm_campaign=Brand&utm_term=gravwell&utm_medium=ppc&utm_source=adwords&hsa_grp=131665588235&hsa_cam=15507596878&hsa_mt=e&hsa_net=adwords&hsa_ver=3&hsa_acc=7435514995&hsa_ad=593346614561&hsa_src=g&hsa_tgt=kwd-299650209089&hsa_kw=gravwell&gad_source=1&gbraid=0AAAAAC2g-OokhNbvrG2lngtGGyQGckmIJ&gclid=EAIaIQobChMIgtSMgOz4hwMVrTUIBR3K8wjHEAAYASAAEgKnzPD_BwE
